README.md

```diff
@@ -7,3 +7,216 @@ Status](https://github.com/maltezfaria/DataFlowTasks.jl/workflows/CI/badge.svg)]
 [![codecov](https://codecov.io/gh/maltezfaria/DataFlowTasks.jl/branch/main/graph/badge.svg?token=UOWU691WWG)](https://codecov.io/gh/maltezfaria/DataFlowTasks.jl)
 ![Lifecycle](https://img.shields.io/badge/lifecycle-experimental-blue.svg)
 
+`DataFlowTasks.jl` is a Julia package dedicated to parallel programming on multi-core shared memory CPUs. From user annotations (READ, WRITE, READWRITE) on program data, `DataFlowTasks.jl` automatically infers dependencies between parallel tasks.
+
+`DataFlowTasks.jl` is particularly easy to use with programs manipulating Julia
+arrays, but can be extended to work with any custom data structure.
+
+## Installation
+
+```julia
+using Pkg
+Pkd.add("https://github.com/maltezfaria/DataFlowTasks.jl.git")
+```
+
+## Basic Usage
+
+The use of `DataFlowTask`s is intended to be as similar to Julia native `Task`s as possible. The API implements three macros :
+* `@dspawn`
+* `@dtask`
+* `@dasync`
+
+which behave like their `Base` counterparts, except they need additional annotations to specify access modes. This is done with the three macros :
+* `@R`
+* `@W`
+* `@RW`
+
+where, in a function argument or at the beginning of a task block, `@R(A)` implies that A will be in read mode in the function/block.
+
+Let's look at a simple example:
+
+```@example
+using DataFlowTasks
+using GraphViz
+
+DataFlowTasks.reset!()
+write!(X, alpha) = (X .= alpha)
+readwrite!(X) = (X .+= 2)
+n = 1000
+A = ones(n)
+function ex()
+    @dspawn readwrite!(@RW(A))              # 1
+    @dspawn write!(@W(@view A[1:500]), 2)   # 2
+    @dspawn write!(@W(@view A[501:n]), 3)   # 3
+    @dspawn readwrite!(@RW(A))              # 4
+end
```

Comment:
either explicitly run `ex()`, or replace it with a `let` block (or a `begin` block)